<!DOCTYPE html>
<html>
<head>
	<title>Example 5-2 - Handling Error</title>
	<script src="q.js"></script>
	<script>
	/*
		Support in jQuery of the following point:
		Rejected, rejection handler returns a value: a catch clause got the error and handled it
	*/
		function saveToDisk() {
			var deferred = Q.defer();
			    setTimeout(deferred.reject("I am rejecting YOU"), 3000);
			    
			    return deferred.promise;
		}

		saveToDisk()
			.then(function (response) {
				console.log("success handler "+ response);
			}, function (reason) {
				console.log("failure handler "+ reason);
				throw new Error("Throwing error....");

			})
			.catch(function (err) {
				console.log("------------------");
				console.log(err);
			});
	</script>
	</head>
	</html>